service/controller/v25/resource/cpf/create.go

```diff
@@ -42,11 +41,19 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			StackName: aws.String(key.MainHostPostStackName(cr)),
 		}
 
-		_, err = cc.Client.ControlPlane.AWS.CloudFormation.DescribeStacks(i)
+		o, err := cc.Client.ControlPlane.AWS.CloudFormation.DescribeStacks(i)
 		if IsNotExists(err) {
 			// fall through
+
 		} else if err != nil {
 			return microerror.Mask(err)
+
+		} else if len(o.Stacks) != 1 {
+			return microerror.Maskf(executionFailedError, "expected one stack, got %d", len(o.Stacks))
+
+		} else if *o.Stacks[0].StackStatus == cloudformation.StackStatusCreateFailed {
+			return microerror.Maskf(executionFailedError, "expected successful status, got %#q", o.Stacks[0].StackStatus)
```

Comment:
Note that I extended the stack creation error handling to see when the stack has failed to be created at some point. There was a problem with the stack creation and I wasted some time before figuring out it was the stack. That was not obvious. It should be visible now in the logs.